File: Reabilit.Domain/DTOs/ProcedureEventDTO.cs
using Reabilit.Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Reabilit.Domain.DTOs;

public class ProcedureEventDTO
{
    public Guid Id { get; set; }
    public required string Title { get; set; }
    public required string Description { get; set; }
    public ProcedureEventStatus Status { get; set; }
    public DateTime StartsOn { get; set; }
    public PatientDTO? Patient { get; set; }
    public DoctorDTO? Doctor { get; set; }
}

File: Reabilit.UI/src/app/services/responseModels/ProcedureEventDTO.ts
import { DoctorDTO } from "./DoctorDTO"
import { PatientDTO } from "./PatientDTO"

export interface ProcedureEventDTO {
    id: string,
    title: string,
    description: string,
    result: string,
    status: ProcedureEventStatus,
    startsOn: string,
    patient: PatientDTO | null,
    doctor: DoctorDTO
}

export enum ProcedureEventStatus {
    Planned = 0,
    Cancelled = 1,
    Finished = 2
}
